project-codeflare/multi-cluster-app-dispatcher
# Fix creation of AWs/Jobs with same name in different namespaces

# Issue link


Resolves #433 
And resolves #383 

In the `genericresource.go` file there is a [SyncQueueJob function](https://github.com/project-codeflare/multi-cluster-app-dispatcher/blob/7f3eda58a6a3b4bb1e92642ddd2357b5065eed09/pkg/controller/queuejobresources/genericresource/genericresource.go#L196) which is used to create resources inside etcd. In this function, the identification of AppWrappers/Jobs was primarily based on a label (`resourceName`) which had a value of the AW name without taking into account the namespace. As a result, when 2 AWs/Jobs with the same name were created in different namespaces, the 2nd AW/Job was identified as a duplicate.

# What changes have been made

- Changed the `resourceName` label to account for the namespace too.
- Changed the labelSelector and function accordingly.
- Created and added this scenario as part of the `e2e-test`.

# Verification steps



1. Have MCAD running.
2. Create two AWs/Jobs with the same name but in different namespaces.
3. See their pods created respectively, and running.

## Checks
- Testing Strategy
